### ISSUE TITLE
Support for Intel Wireless Daemon (iwd)

### ISSUE THREAD

**Issue by vially** [joukewitteveen/netctl#156]
Are there any plans to add support for the [Intel Wireless Daemon](https://git.kernel.org/pub/scm/network/wireless/iwd.git)?

Recently NetworkManager [added initial support](https://cgit.freedesktop.org/NetworkManager/NetworkManager/commit/?id=24bc95316d7d7c0690c4d808d785785f1ff46c35) for it too.

**Comment by denji** [comment]
https://git.kernel.org/pub/scm/network/wireless/iwd.git/snapshot/iwd-0.1.tar.gz

**Comment by joukewitteveen** [comment]
I was made aware of the release of IWD 0.1. However, there is no sign of a command line client. Such a client is on the [TODO](https://git.kernel.org/pub/scm/network/wireless/iwd.git/tree/TODO) list (at the bottom) and required for support by netctl.

**Comment by aenda** [comment]
Version 0.2 from the Arch repos provides a cli (iwctl) to iwd (described at the [wiki](https://wiki.archlinux.org/index.php/iwd)). Is there still some necessary functionality missing from it? I also am not 100% sure, but believe that the todo list just hasn't been updated in light of iwctl being made available.

**Comment by joukewitteveen** [comment]
Okay, so there is a sign of a command line client. However I cannot find a manpage. Does it even work non-interactively? The tool feels very unfinished...

**Comment by aenda** [comment]
Looking into it, it seems that iwctl is for interactive use only, and otherwise the daemon can be managed via dbus, but there isn't much documentation for that case I can find besides the wiki page.

**Comment by cosmojg** [comment]
Just a heads-up, iwctl is usable non-interactively by simply appending commands (e.g. iwctl station wlp2s0 connect myssid).

**Comment by thithib** [comment]
I think this should be reopened now, since `iwd` is finally mature enough to have become the recommended way of managing Wi-Fi connections on Arch Linux instead of `wpa_supplicant` and to have been included in the archiso.

I love `netctl` and I've been using it for a very long time now, but to me it's losing much of its appeal for as long as it doesn't support `iwd`.

**Comment by cosmojg** [comment]
I agree with the above. I think it makes a lot of sense to start fielding pull requests for iwd support at this point. Is there any reason we can't reopen this?

**Comment by joukewitteveen** [comment]
While the situation is continually improving, I still think iwd is doing too much and is not a 'do one thing and do it well' tool. This makes integration in netctl hard (see rant below). I am open to patches, though, and willing to help out where needed. However, until `iwd` has a real advantage over `wpa_supplicant`, I will not spend my time spearheading support for `iwd` myself.

_Rant_:
Despite widespread sentiment, netctl is a capable and effective implementation of systemd-aware profile-based network configuration with maximum flexibility and control. In comparison, `iwd` is less flexible as it tries to take networking control out of the hands of the user and into its own hands. For many (probably a majority), this will be fine (`netctl-auto` exists for a reason), but conceptually, these `automagic` tools do not align well with netctl. A similar situation occurs around the internal dhcp client of systemd. Supporting it in netctl would be easy if only systemd provided a cli. Likewise, everything `ifplugd` does is implemented in systemd, but it is only hooked up internally and therefore unusable for third party tools.
